lib/src/component/error_boundary_mixins.dart

```diff
@@ -260,23 +263,28 @@ mixin ErrorBoundaryMixin<T extends ErrorBoundaryPropsMixin, S extends ErrorBound
   // ---------------------------------------------- /\ ----------------------------------------------
 
   String _domAtTimeOfError;
-  /*Error||Exception*/dynamic _lastError;
-  ReactErrorInfo _lastErrorInfo;
+  List<String> _errorLog = [];
+  List<ReactErrorInfo> _callStackLog = [];
   Timer _identicalErrorTimer;
 
   /// Called by [componentDidCatch].
   void _handleErrorInComponentTree(/*Error||Exception*/dynamic error, ReactErrorInfo info) {
     // ----- [1] ----- //
     if (props.fallbackUIRenderer != null) {
-      _lastError = error;
-      _lastErrorInfo = info;
+      _errorLog.add(error.toString());
+      _callStackLog.add(info);
       _logErrorCaughtByErrorBoundary(error, info); // [3]
       return;
     }
     // ----- [2] ----- //
     else {
-      bool sameErrorWasThrownTwiceConsecutively =
-          error.toString() == _lastError?.toString() && info.componentStack == _lastErrorInfo.componentStack;
+      bool sameErrorWasThrownTwiceConsecutively = false;
+
+      for (var i = 0; i < _errorLog.length; i++) {
+        if (_errorLog[i] == error.toString() && _callStackLog[i].componentStack == info.componentStack) {
+          sameErrorWasThrownTwiceConsecutively = true;
```

Comment:
#nit this could short circuit as soon as it finds a match
```suggestion
          sameErrorWasThrownTwiceConsecutively = true;
          break;
```